### PR TITLE
trying to build 0.18rc with rc tag on conda-forge channel

### PR DESCRIPTION
**Pull request by amueller**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,12 +33,12 @@ install:
       bash $MINICONDA_FILE -b
 
       export PATH=/Users/travis/miniconda3/bin:$PATH
-      conda config --add channels conda-forge
+      conda config --add channels ['conda-forge']
```
Comment (jakirkham) [inline .travis.yml]: That's fun. 😕

File: conda-forge.yml [hunk 2]
```diff
@@ -4,3 +4,8 @@ travis:
 appveyor:
   secure:
     BINSTAR_TOKEN: MP4hZYylDyUWEsrt3u3cod2sbFeRwUziH02mvQOdbjsTO/l1yIxDkP/76rSIjcGC
+channels:
+  sources:
+  - [conda-forge]
```
Comment (jakirkham) [inline conda-forge.yml]: Maybe drop the `[` `]`

File: .travis.yml (hunk 1 above)
Comment (amueller) [inline .travis.yml]: ok ^^

File: ci_support/run_docker_build.sh [hunk 3]
```diff
@@ -13,7 +13,7 @@ docker info
 config=$(cat <<CONDARC
 
 channels:
- - conda-forge
+ - ['conda-forge']
```
Comment (jakirkham) [inline ci_support/run_docker_build.sh]: With quotes too. 😕 That's fun.

File: conda-forge.yml (hunk 2 above)
Comment (jakirkham) [inline conda-forge.yml]: Actually we should be able to drop this `sources` section as it is default.

Comment (jakirkham) [inline conda-forge.yml]: Raised this issue upstream ( https://github.com/conda-forge/conda-smithy/issues/306 ).

**Comment by jakirkham** [comment]
So this is set to go into `master` currently. Am going to add a new branch for the `rc`. We will need to change the base branch and resolve conflicts potentially, but it shouldn't be bad.

**Comment by jakirkham** [comment]
Ok, added an `rc` branch and changed this over to it. Now we should be good to go.

File: conda-forge.yml [hunk 4]
```diff
@@ -4,3 +4,8 @@ travis:
 appveyor:
   secure:
     BINSTAR_TOKEN: MP4hZYylDyUWEsrt3u3cod2sbFeRwUziH02mvQOdbjsTO/l1yIxDkP/76rSIjcGC
+channels:
+  sources:
+  - conda-forge
```
Comment (jakirkham) [inline conda-forge.yml]: So I think we can just drop `sources`. This is default anyways.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{% set version = "0.17.1" %}
+{% set version = "0.18rc" %}
```
Comment (jakirkham) [inline recipe/meta.yaml]: Is there a number associated with the RC (e.g. `0.18rc1`)?

**Comment by jakirkham** [comment]
Couple comments. Should be easy to address. Otherwise LGTM.

**Comment by ogrisel** [comment]
note that 0.18rc has a broken test under Python 3 that has been fixed in 0.18rc1. FYI I am working on 0.18rc2 with another important bugfix so please do not publish the package for 0.18rc or 0.18rc1.

**Comment by jakirkham** [comment]
Ok, thanks. Noted.

**Comment by amueller** [comment]
@jakirkham so would people that subscribed to the matplotlib rc also get this?
Or do people usually only specify a particular package with a label?
If no-one gets this one without explicitly saying so, I feel like we shouldn't worry too much.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -8,8 +8,8 @@ package:
 
 source:
   fn: scikit-learn-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/s/scikit-learn/scikit-learn-{{ version }}.tar.gz
-  md5: a2f8b877e6d99b1ed737144f5a478dfc
+  url: https://github.com/scikit-learn/scikit-learn/archive/{{ version }}.tar.gz
+  md5: 3cb6b237bcfb77168fb0b41c4cb9cff0
```
Comment (jakirkham) [inline recipe/meta.yaml]: Checksum changed. Though may not be worth bothering with until we have the RC we want.

**Comment by jakirkham** [comment]
> @jakirkham so would people that subscribed to the matplotlib rc also get this?

Yep, unfortunately. Though it is better than putting it in `main`.

There was some discussion of having a flag for `conda` to control this behavior. See this issue ( https://github.com/conda/conda/issues/3125 ).

The CFEP (which is not yet accepted) suggests possibly including the package name in the label like [so]( https://github.com/conda-forge/conda-forge-enhancement-proposals/pull/3/files#diff-829e88916d98f013f67ccc03e894903cR75 ). This would resolve the problem, but is a bit verbose and feels a bit unfriendly if one starts installing lots of dev packages. Still it may be the best available option.

Not really sure which works best and all of them are a little incomplete. Would be curious to hear your thoughts/preferences.

> Or do people usually only specify a particular package with a label?

Had suggested a [mechanism]( https://github.com/conda-forge/conda-forge.github.io/issues/147#issuecomment-224597202 ) for doing something like that as well, but it is a bit inelegant. It could also suffer from issues if a new dependency is added.

> If no-one gets this one without explicitly saying so, I feel like we shouldn't worry too much.

Agreed. I'm not too worried about it either.

Simply having the user make the choice is the most important. The rest is questions about the granularity of that choice and its impact.

**Comment by ogrisel** [comment]
Arlgl:

```
.............................................................
command ./ci_support/run_docker_build.sh took more than 120 minutes to run
```

**Comment by amueller** [comment]
Sooo... disable the tests ? lol

**Comment by jakirkham** [comment]
There are other options. Will look at it more when I'm back in front of a computer (on phone now).

**Pull request closed without merging**